docs/pages/tutorials/minimal/use-onchain.mdx

```diff
@@ -0,0 +1,328 @@
+
+```solidity
+contract ScriptClient is Script {
+  IWorld world;
+
+  function run() external {
+```
+
+This is the function that runs the script.
+
+```solidity
+    // Load the private key from the `PRIVATE_KEY` environment variable (in .env)
+    uint256 myPrivateKey = vm.envUint("PRIVATE_KEY");
+
+    // Start broadcasting transactions from that account
+    vm.startBroadcast(myPrivateKey);
+```
+
+Read the private key from the environment (which includes the content of the `.env` file) and use it to issue transactions.
+
+```solidity
+    // Connect to the world using `WORLD_ADDR` from .env
+    console.log("Connecting to world at ", vm.envAddress("WORLD_ADDR"));
+    world = IWorld(vm.envAddress("WORLD_ADDR"));
+```
+
+Connect to the `World` at `WORLD_ADDR`.
+
+```solidity
+    console.log("The counter values is now ", Counter.get(world));
+
+    console.log("Incrementing the counter");
+    world.increment();
+
+    console.log("The counter values is now ", Counter.get(world));
+```
+
+This is the actual script.
+Note that it has access to `Counter.get(world)` to read the data, but to modify it you have to go through a `System`.
+Remember, _there are no secrets on the blockchain_ and therefore we can assume that anybody can read anything at any time after it is posted - so we might as well provide read access directly.
+
+<details>
+
+<summary>Try to bypass security</summary>
+
+There is another script `BypassSecurity`, which attempts to bypass the `System` logic and set `Counter` directly.
+You can run it with this command.
+
+```sh copy
+forge script --rpc-url http://localhost:8545 --broadcast script/ScriptClient.s.sol:BypassSecurity
+```
+
+It fails with this error:
+
+```
+== Logs ==
+  Connecting to world at  0x5FbDB2315678afecb367f032d93F642f64180aa3
+  Just before trying to reset the counter
+Error:
+e64c0348:ROOT_NAMESPACE/Counter
+```
+
+This is the relevant portion of the code.
+
+```solidity
+    // Try to set `Counter` directly. See that it fails.
+    console.log("Just before trying to reset the counter");
+    Counter.set(world, 0);
+    console.log("Just after trying to reset the counter");
+```
+
+We know the the failure happens in the `Counter.set(world, 0);` line because we see the log entry for the line before it, but not the one for the line immediately after it.
+
+</details>
+
+```solidity
+    // Stop broadcasting transactions from te account
+    vm.stopBroadcast();
+  }
+}
+```
+
+Finally, stop broadcasting transactions using the private key.
+
+</details>
+
+## Use the `World` from a contract
+
+You can use the contract [in the examples directory](https://github.com/latticexyz/mud/tree/main/examples/onchain-client).
+
+1. Read the environment variables.
+
+   ```sh copy
+   . .env
+   ```
+
+1. Deploy the sample contact.
+
+   ```sh copy
+   forge create OnchainClient --private-key $PRIVATE_KEY --constructor-args $WORLD_ADDR
+   ```
+
+1. Set `CLIENT_ADDR` to the address to which the contract is deployed.
```

Comment:
Where should they set this? In the `.env` file? (Also let's use `CLIENT_ADDRESS`)